.github/workflows/test.yml

```diff
@@ -51,23 +52,24 @@ jobs:
           sudo apt-get update
           sudo apt-get install -y --no-install-recommends vagrant vagrant-libvirt libvirt-clients libvirt-daemon-system libvirt-daemon qemu-system-x86 qemu-utils dnsmasq
           sudo chmod 666 /var/run/libvirt/libvirt-sock
-      - name: Install Ansible
-        run: pip install --upgrade ansible-core
-      - name: Install dependencies
-        run: pip install -r requirements.txt
-      - name: Install collection dependencies
-        run: ansible-galaxy install -r requirements.yml
+      - name: Setup environment
+        run: |
+          ./setup-environment
+          source .venv/bin/activate
+          echo "$VIRTUAL_ENV/bin" >> $GITHUB_PATH
+          echo "VIRTUAL_ENV=$VIRTUAL_ENV" >> $GITHUB_ENV
```

Comment:
I don't think you need to setup a virtualenv with `setup-python`. Also, see https://github.com/actions/setup-python?tab=readme-ov-file#caching-packages-dependencies for how to use the caching.

Reply:
I'm not using `setup-python` unless I'm missing something?

Reply:
line 44:
```
      - name: Set up Python
        uses: actions/setup-python@v5
```